src/lightning/pytorch/utilities/combined_loader.py

```diff
@@ -172,6 +191,14 @@ class CombinedLoader(Iterable):
         {'a': tensor([0, 1, 2, 3]), 'b': tensor([0, 1, 2, 3, 4])}
         {'a': tensor([4, 5]), 'b': tensor([5, 6, 7, 8, 9])}
         {'a': tensor([0, 1, 2, 3]), 'b': tensor([10, 11, 12, 13, 14])}
+        >>> combined_loader = CombinedLoader(iterables, 'max_size')
+        >>> len(combined_loader)
+        3
+        >>> for batch in combined_loader:
+        ...     print(batch)
+        {'a': tensor([0, 1, 2, 3]), 'b': tensor([0, 1, 2, 3, 4])}
+        {'a': tensor([4, 5]), 'b': tensor([5, 6, 7, 8, 9])}
+        {'a': None, 'b': tensor([10, 11, 12, 13, 14])}
```

Comment:
Love it. Do you see any problems where a dataset returns None vs. None from the max_size limit being indistinguishable?  That's pretty rare/unusual, but a way out would be to simply drop the key. The difference for the end user would simply be that they check whether the key is in the dict vs. checking that the value is None.

Reply:
Seems like a very niche thing to do (since `default_collate` does not support `None`), that could be worked around, by using a sentinel instead of None if the end user really needs the distinction.
Also, as I commented in the issue, distinction by dropping keys would be possible only with `dict`, while `list`, `tuple` and `namedtuple` would have to use `None` anyway, which might make things more confusing for the users.

That being said, I can add this functionality if you think it's a good addition.

Reply:
That sounds good. I think it is actually a great suggestion to use a sentinel value.